server/py/services/api/crud/secrets.py

```diff
@@ -419,6 +422,112 @@ def get_project_secret(
     def is_internal_project_secret_key(self, key: str) -> bool:
         return key.startswith(self.internal_secrets_key_prefix)
 
+    def store_secret_tokens(
+        self,
+        secret_tokens: list[mlrun.common.schemas.SecretToken],
+        authenticated_user_id: str,
+        authenticated_username: str,
+    ) -> mlrun.common.schemas.StoreSecretTokensResponse:
+        """
+        Validate and store offline tokens as Kubernetes secrets.
+
+        :param secret_tokens: List of SecretToken objects to store.
+        :param authenticated_user_id: User ID to validate token ownership.
+        :param authenticated_username: Username used to name Kubernetes secrets.
+        :return: StoreSecretTokensResponse object with created, updated, and skipped tokens.
+        """
+        if not secret_tokens:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "Failed to store secret tokens – no tokens provided"
+            )
+
+        valid_tokens = self._validate_and_decode_offline_tokens(
+            secret_tokens, authenticated_user_id
+        )
+
+        # TODO: move init iguazio_client
+        iguazio_client = framework.utils.clients.iguazio.v4.Client()
+        iguazio_client.refresh_access_tokens(secret_tokens)
+
+        created_tokens, updated_tokens, skipped_tokens = [], [], []
+
+        for token_name, decoded_token in valid_tokens.items():
+            action = self.secrets_provider.create_or_update_user_token_secret(
+                username=authenticated_username,
+                token_name=token_name,
+                token=decoded_token["raw_token"],
+                expiration=decoded_token["exp"],
+            )
+            if action == mlrun.common.schemas.SecretEventActions.created:
+                created_tokens.append(token_name)
+            elif action == mlrun.common.schemas.SecretEventActions.updated:
+                updated_tokens.append(token_name)
+            elif action == mlrun.common.schemas.SecretEventActions.skipped:
+                skipped_tokens.append(token_name)
+
+        return mlrun.common.schemas.StoreSecretTokensResponse(
+            created_tokens=created_tokens,
+            updated_tokens=updated_tokens,
+            skipped_tokens=skipped_tokens,
+        )
```

Comment:
nitpicking - can be simplified a bit
```suggestion
  token_actions = defaultdict(list)
  
  for token_name, token_data in valid_tokens.items():
      action = self.secrets_provider.create_or_update_user_token_secret(
          username=authenticated_username,
          token_name=token_name,
          token=token_data["raw_token"],
          expiration=token_data["exp"],
      )
      token_actions[action].append(token_name)
  
  return StoreSecretTokensResponse(
      created_tokens=token_actions[SecretEventActions.created],
      updated_tokens=token_actions[SecretEventActions.updated],
      skipped_tokens=token_actions[SecretEventActions.skipped],
  )

```